app/controllers/taxa/names_controller.rb

```diff
@@ -200,4 +204,16 @@ def parse_classification_controller_params
     @move_to = params[:move_to]
   end
 
+  def limit_names
+    if current_user.newish? || (current_user.newish? && current_user.assistant_curator?)
+       names_count = Curator.total_objects_curated_by_action_and_user(Activity.add_common_name.id, current_user.id, [ChangeableObjectType.synonym.id],
+                             return_type = 'count', created_at = DateTime.now.to_date.beginning_of_day..DateTime.now.to_date.end_of_day)
+       if names_count == 0
+          return false
+       else
+         return true
+       end
+    end
+    return false
```

Comment:
Indeed, you don't need the if block, here, and you don't need the last "return". Something like

```
if blah blah
  return names_count != 0
end
false
```